### PR TITLE
Ruby 3.3.6 update

### PR DESCRIPTION
**Pull request by thejonroberts**
### What github issue is this PR for, if any?
No issue

### What changed, and _why_?
Update ruby version

### How is this **tested**? (please write tests!) 💖💪
Current specs only

### CODE REVIEW

File: app/services/followup_service.rb [hunk 1]
```diff
@@ -16,9 +16,7 @@ def self.create_followup(case_contact, creator, note)
     followup
   end
 
-  private_class_method
-
-  def self.send_notification(followup, creator)
+  private_class_method def self.send_notification(followup, creator)
```
Comment (thejonroberts) [inline app/services/followup_service.rb:19]: this  fixed a warning: private_class_method is ignored unless called with a symbol, like:
`private_class_method :send_notification`
or used this way.

File: app/views/case_contacts/form/details.html.erb [hunk 2]
```diff
@@ -54,7 +54,6 @@
           </h3>
           <% min_date = CaseContact::MINIMUM_DATE %>
           <% current_date = Time.zone.today %>
-          <% initial_value = @case_contact.occurred_at&.to_date %>
```
Comment (thejonroberts) [inline app/views/case_contacts/form/details.html.erb:57]: unused (warning)

File: spec/services/placement_export_csv_service_spec.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@
     casa_org = create(:casa_org, name: "Fake Name", display_name: "Fake Display Name")
     placement_type = create(:placement_type, casa_org: casa_org)
     creator = create(:user)
-    placement = create(:placement, creator: creator, placement_type: placement_type) # rubocop:disable Lint/UselessAssignment
```
Comment (thejonroberts) [inline spec/services/placement_export_csv_service_spec.rb:9]: unused var (warning)

File: spec/support/capybara.rb [hunk 4]
```diff
@@ -15,14 +15,7 @@
...
 Capybara::Screenshot.autosave_on_failure = true
-
-module Capybara
-  module Screenshot
-    def self.capybara_tmp_path
-      Rails.root.join("tmp", "screenshots#{ENV["GROUPS_UNDERSCORE"]}")
-    end
-  end
-end
+Capybara.save_path = Rails.root.join("tmp", "screenshots#{ENV["GROUPS_UNDERSCORE"]}")
```
Comment (thejonroberts) [inline spec/support/capybara.rb:18]: method redefined (warning). I'm like 90% sure this is equivalent, but may want to check CI artifacts work.

File: app/views/case_contacts/form/details.html.erb (hunk 2 above)
Comment (compwron) [reply app/views/case_contacts/form/details.html.erb:57]: it's an erb, so *probably* used by js somewhere?

**Comment by compwron** [comment]
has merge conflicts now sorry :/

File: app/views/case_contacts/form/details.html.erb (hunk 2 above)
Comment (thejonroberts) [reply app/views/case_contacts/form/details.html.erb:57]: No, but I only know that because I wrote it haha. I left this straggler.

**Comment by thejonroberts** [comment]
@compwron no problem, rebased now.

**Pull request merged**